gammapy/background/models.py

```diff
@@ -71,3 +82,724 @@ def evaluate(self, x, y):
+            https://github.com/gammapy/gammapy/pull/292#discussion_r33843508
+
+        Parameters
+        ----------
+        energy : `~astropy.units.Quantity`
+            energy to search for
+
+        Returns
+        -------
+        bin_pos : `~int`
+            index of the energy bin containing the specified energy
+        bin_edges : `~astropy.units.Quantity`
+            energy bin edges [E_min, E_max)
+        """
+        # check shape of energy: only 1 value is accepted
+        nvalues = len(energy.flatten())
+        if nvalues != 1:
+            raise IndexError("Expected exactly 1 value for energy, got {}."
+                             .format(nvalues))
+
+        # check that the specified energy is within the boundaries of the model
+        energy_extent = self.spectrum_extent
+        if not (energy_extent[0] <= energy) and (energy < energy_extent[1]):
+            ss_error = "Specified energy {}".format(energy)
+            ss_error += " is outside the boundaries {}.".format(energy_extent)
+            raise ValueError(ss_error)
+
+        energy_edges_low = self.energy_bins[:-1]
+        energy_edges_high = self.energy_bins[1:]
+        bin_pos = np.searchsorted(energy_edges_high, energy)
+        bin_edges = Quantity([energy_edges_low[bin_pos], energy_edges_high[bin_pos]])
+
+        return bin_pos, bin_edges
+
+    def plot_image(self, energy, ax=None, style_kwargs=None):
+        """Plot image for the energy bin containing the specified energy.
+
+        Parameters
+        ----------
+        energy : `~astropy.units.Quantity`
+            energy of bin to plot the bg model
+        ax : `~matplotlib.axes.Axes`, optional
+            axes of the figure for the plot
+        style_kwargs : `~dict`, optional
+            style options for the plot
+
+        Returns
+        -------
+        ax : `~matplotlib.axes.Axes`
+            axes of the figure containing the plot
+        """
+        import matplotlib.pyplot as plt
+        from matplotlib.colors import LogNorm
+        from mpl_toolkits.axes_grid1 import make_axes_locatable
+
+        energy = energy.flatten() # flatten
+        # check shape of energy: only 1 value is accepted
+        nvalues = len(energy)
+        if nvalues != 1:
+            ss_error = "Expected exactly 1 value for energy, got {}.".format(nvalues)
+            raise IndexError(ss_error)
+        else:
+            energy = Quantity(energy[0])
+
+        extent = self.image_extent
+        energy_bin_centers = self.energy_bin_centers
+
+        # find energy bin containing the specified energy
+        energy_bin, energy_bin_edges = self.find_energy_bin(energy)
+        ss_energy_bin_edges = "[{0}, {1}) {2}".format(energy_bin_edges[0].value,
+                                                      energy_bin_edges[1].value,
+                                                      energy_bin_edges.unit)
+
+        # get data for the plot
+        ii = energy_bin
```

Comment:
There's no point in creating a second variable `ii` that is the same as `energy_bin`, right?
Just use `energy_bin`?